digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -440,7 +440,7 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
           logger.debug(s"Reserve $reservation")
           val addressesOnPart = RoadAddressDAO.fetchByRoadPart(reservation.roadNumber, reservation.roadPartNumber, false)
           val mapping = roadLinkService.getViiteRoadLinksByLinkIdsFromVVH(addressesOnPart.map(_.linkId).toSet, false, frozenTimeVVHAPIServiceEnabled)
-            .map(rl => rl.linkId -> RoadAddressLinkBuilder.getRoadType(rl.administrativeClass, rl.linkType)).toMap
+            .map(rl => rl.linkId -> addressesOnPart.find(_.linkId == rl.linkId).head.roadType).toMap
```

Comment:
Mapping becomes obsolete because it can be directly read from the address information. Besides finding by link id may map to incorrect road address object (there may be more than one on link) and should use mapping by road address id.(project_link.road_address_id = road_address.id in SQL)

Reply:
line 423:    def toProjectLink(roadTypeMap: Map[Long, RoadType])(roadAddress: RoadAddress): ProjectLink = {
should drop that mapping and just do
line 428:        LinkStatus.NotHandled, roadTypeMap.getOrElse(roadAddress.linkId, RoadType.Unknown), roadAddress.linkGeomSource, GeometryUtils.geometryLength(roadAddress.geometry), roadAddress.id, roadAddress.ely, reversed = false)
->
line 428:        LinkStatus.NotHandled, roadAddress.roadType, roadAddress.linkGeomSource, GeometryUtils.geometryLength(roadAddress.geometry), roadAddress.id, roadAddress.ely, reversed = false)

Reply:
Saw another bug in here, should instead of 
GeometryUtils.geometryLength(roadAddress.geometry)
have 
(roadAddress.endMValue - roadAddress.startMValue) because geometry is simplified in road address data.